lib/devicetrust/native/device_windows.go

```diff
@@ -0,0 +1,489 @@
+		return nil, trace.BadParameter("no endorsement keys found in tpm")
+	}
+	// TODO(noah): Marshal EK Certificate instead of key if present:
+	// https://github.com/gravitational/teleport.e/issues/1393
+	encodedEK, err := x509.MarshalPKIXPublicKey(eks[0].Public)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return encodedEK, nil
+}
+
+// loadOrCreateAK attempts to load an AK from disk. A NotFound error will be
+// returned if no such file exists.
+func loadAK(
+	tpm *attest.TPM,
+	persistencePath string,
+) (*attest.AK, error) {
+	ref, err := os.ReadFile(persistencePath)
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+
+	ak, err := tpm.LoadAK(ref)
+	if err != nil {
+		return nil, trace.Wrap(err, "loading ak into tpm")
+	}
+
+	return ak, nil
+}
+
+func createAndSaveAK(
+	tpm *attest.TPM,
+	persistencePath string,
+) (*attest.AK, error) {
+	ak, err := tpm.NewAK(nil)
+	if err != nil {
+		return nil, trace.Wrap(err, "creating ak")
+	}
+
+	// Write it to the well-known location on disk
+	ref, err := ak.Marshal()
+	if err != nil {
+		return nil, trace.Wrap(err, "marshalling ak")
+	}
+	err = os.WriteFile(persistencePath, ref, 600)
+	if err != nil {
+		return nil, trace.Wrap(err, "writing ak to disk")
+	}
+
+	return ak, nil
+}
+
+func enrollDeviceInit() (*devicepb.EnrollDeviceInit, error) {
+	akPath, err := setupDeviceStateDir(os.UserHomeDir)
+	if err != nil {
+		return nil, trace.Wrap(err, "setting up device state directory")
+	}
+
+	tpm, err := openTPM()
+	if err != nil {
+		return nil, trace.Wrap(err, "opening tpm")
+	}
+	defer tpm.Close()
+
+	// Try to load an existing AK in the case of re-enrollment, but, if the
+	// AK does not exist, create one and persist it.
+	ak, err := loadAK(tpm, akPath)
+	if err != nil {
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err, "loading ak")
+		}
+		log.Debug("No existing AK was found on disk, an AK will be created.")
+		ak, err = createAndSaveAK(tpm, akPath)
+		if err != nil {
+			return nil, trace.Wrap(err, "creating ak")
+		}
+	} else {
+		log.Debug("Existing AK was found on disk, it will be reused.")
+	}
+	defer ak.Close(tpm)
+
+	deviceData, err := collectDeviceData()
+	if err != nil {
+		return nil, trace.Wrap(err, "collecting device data")
+	}
+
+	marshaledEK, err := getMarshaledEK(tpm)
+	if err != nil {
+		return nil, trace.Wrap(err, "marshalling ek")
+	}
+
+	credentialID, err := credentialIDFromAK(ak)
+	if err != nil {
+		return nil, trace.Wrap(err, "determining credential id")
+	}
+
+	return &devicepb.EnrollDeviceInit{
+		CredentialId: credentialID,
+		DeviceData:   deviceData,
+		Tpm: &devicepb.TPMEnrollPayload{
+			Ek: &devicepb.TPMEnrollPayload_EkKey{
+				EkKey: marshaledEK,
+			},
+			AttestationParameters: devicetrust.AttestationParametersToProto(
+				ak.AttestationParameters(),
+			),
+		},
+	}, nil
+}
+
+// credentialIDFromAK produces a deterministic short-ish unique-ish human
+// readable-ish string identifier for a given AK. This can then be used as a
```

Comment:
```suggestion
// credentialIDFromAK produces a deterministic, short-ish, unique-ish, human
// readable-ish, string identifier for a given AK. This can then be used as a
```